src/hotspot/share/runtime/deoptimization.cpp

```diff
@@ -748,9 +749,14 @@ Deoptimization::UnrollBlock* Deoptimization::fetch_unroll_info_helper(JavaThread
   }
 #endif
 
+  int caller_actual_parameters = -1; // value not used except for interpreted frames, see below
+  if (deopt_sender.is_interpreted_frame()) {
+    caller_actual_parameters = callee_parameters + (caller_was_method_handle ? 1 : 0);
```

Comment:
Previously, if caller_was_method_handle was set, we would pass in 0 below, which was wrong for the has_member_arg case, and I suspect it broke JVMTI PopFrame for platforms that don't use popframe_move_outgoing_args, but I don't have a test for this suspicion.